Sources/Workspace/WorkspaceState.swift

```diff
@@ -154,6 +154,11 @@ fileprivate struct WorkspaceStateStorage {
         struct Container: Codable {
             var dependencies: [Dependency]
             var artifacts: [Artifact]
+
+            init(dependencies: [Dependency], artifacts: [Artifact]) {
+                self.dependencies = dependencies.sorted { $0.packageRef.identity < $1.packageRef.identity }
```

Comment:
Sorting these seems like a good idea, but since this constructor takes ordered inputs already, does it make sense to sort them here?  I would expect it to be done at wherever the Dictionary is getting iterated over (admittedly I don't know this code so it could be that I'm missing something there, but in general, for any type that takes ordered initialization parameters, I would expect order to be preserved).

Reply:
Ah, good point.  I did it here because it clearly provides the stable serialization guarantee regardless of what the caller does, but in fact there's only one caller, a few lines up.  I'll update the PR accordingly.

I also considered changing the underlying Collection conformance extensions to provide stable ordering, but that seemed broader than what we need.

Reply:
I think doing this in the serialization is the right place, since the focus on consistency of the serialization for, not the external data mode 

but I would love to see a unit test :D